src/hooks/useTasks.js

```diff
@@ -15,67 +15,75 @@ export default function useTasks() {
+          if (isVotingTaskOpen(incompleteTasks[i].jurors[j], currentPhase)) {
+            openTasks.push({
+              number: incompleteTasks[i].number,
+              state: incompleteTasks[i].state,
+              createdAt: parseInt(incompleteTasks[i].createdAt, 10) * 1000,
+              juror: incompleteTasks[i].jurors[j].juror.id,
+              disputeId: incompleteTasks[i].dispute.id,
+              commitment: incompleteTasks[i].jurors[j].commitment,
+              outcome: incompleteTasks[i].jurors[j].outcome,
+              phase: getTaskName(currentPhase),
+              phaseType: currentPhase,
+              dueDate: nextTransition,
+            })
+          }
+        }
+      } else {
+        if (isAppealTaskOpen(incompleteTasks[i], currentPhase)) {
+          // We are in appeal or confirm and only need to generate a single task
+          // (rather than one per juror) if the task is still open
           openTasks.push({
-            number: tasks[i].number,
-            state: tasks[i].state,
-            createdAt: parseInt(tasks[i].createdAt, 10) * 1000,
+            number: incompleteTasks[i].number,
+            state: incompleteTasks[i].state,
+            createdAt: parseInt(incompleteTasks[i].createdAt, 10) * 1000,
             juror: 'Anyone',
-            disputeId: tasks[i].dispute.id,
-            phase: getTaskName(currentRoundPhases[i].phase),
-            dueDate: currentRoundPhases[i].nextTransition,
-            phaseType: currentRoundPhases[i].phase,
-            open: isAppealTaskOpen(tasks[i], currentRoundPhases[i].phase),
+            disputeId: incompleteTasks[i].dispute.id,
+            phase: getTaskName(currentPhase),
+            dueDate: nextTransition,
+            phaseType: currentPhase,
           })
-        } else {
-          for (let j = 0; j < tasks[i].jurors.length; j++) {
-            if (currentRoundPhases[i].phase !== DisputesTypes.Phase.Ended) {
-              openTasks.push({
-                number: tasks[i].number,
-                state: tasks[i].state,
-                createdAt: parseInt(tasks[i].createdAt, 10) * 1000,
-                juror: tasks[i].jurors[j].juror.id,
-                disputeId: tasks[i].dispute.id,
-                commitment: tasks[i].jurors[j].commitment,
-                outcome: tasks[i].jurors[j].outcome,
-                phase: getTaskName(currentRoundPhases[i].phase),
-                phaseType: currentRoundPhases[i].phase,
-                dueDate: currentRoundPhases[i].nextTransition,
-                open: isVotingTaskOpen(
-                  tasks[i].jurors[j],
-                  currentRoundPhases[i].phase
-                ),
-              })
-            }
-          }
         }
       }
     }
     return openTasks
-  }, [currentRoundPhasesKey, tasks]) // eslint-disable-line react-hooks/exhaustive-deps
+  }, [convertedTasksPhasesKey, tasks]) // eslint-disable-line react-hooks/exhaustive-deps
```

Comment:
Let's add a comment explaining that we're using our own generated cache key and don't need to add `convertedTasks` to the dependency list.